spec/features/documents/report_spec.rb

```diff
@@ -147,15 +135,29 @@
       expect(page).to have_selector "a.launch-button[href='#{url2}']"
       expect(page).to have_selector "a.launch-button[href='#{url3}']"
     end
+    describe 'multiple users with the same username' do
+      scenario 'report should show report for "bar" runKey' do
+        url1 = doc_url(server, {recordid: dup_student1_doc.id, documentServer: 'https://www.example.com/', runKey: 'bar'})
+        url2 = doc_url(server, {recordid: dup_student2_doc.id, documentServer: 'https://www.example.com/', runKey: 'foo'})
+        visit report_path(owner: author.username, recordname: template.title, server: server, reportUser: dup_student1.username, runKey: 'bar')
+        expect(page).to have_selector('.launch-button', count: 1)
+        expect(page).to have_selector "a.launch-button[href='#{url1}']"
+      end
+      scenario 'report should show report for "foo" runKey' do
+        url1 = doc_url(server, {recordid: dup_student1_doc.id, documentServer: 'https://www.example.com/', runKey: 'bar'})
+        url2 = doc_url(server, {recordid: dup_student2_doc.id, documentServer: 'https://www.example.com/', runKey: 'foo'})
+        visit report_path(owner: author.username, recordname: template.title, server: server, reportUser: dup_student2.username, runKey: 'foo')
+        expect(page).to have_selector('.launch-button', count: 1)
+        expect(page).to have_selector "a.launch-button[href='#{url2}']"
+      end
+    end
```

Comment:
I'm not sure why it is necessary, but the `:owner` is used to help look up the master/original document:
https://github.com/concord-consortium/document-store/blob/improve-report-2/app/controllers/documents_controller.rb#L325